### PR TITLE
Improve packagecode and other release prep

### PR DESCRIPTION
**Pull request by pombredanne**
This PR adds:
- a few improvements and bug fixes to package processing on Python and Gradle
- fix the Apple M1 installation issues

This is a potpourri towards the latest release.

### Tasks

* [x] Reviewed [contribution guidelines](https://github.com/nexB/scancode-toolkit/blob/develop/CONTRIBUTING.rst)
* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] Tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR
  Run [tests](https://scancode-toolkit.readthedocs.io/en/latest/contribute/contrib_dev.html#running-tests) locally to check for errors. 
* [x] Commits are in uniquely-named feature branch and has no merge conflicts 📁

### CODE REVIEW

**Comment by pombredanne** [comment]
Closing now, while fixing flaky tests

**Comment by pombredanne** [comment]
@JonoYang please have a look at this before I merge ... This is a bit of a mini mess of many unrelated things:
- improved gradle parsing
- improved setup.py parsing
- fix for the PyPI heisenbug 
- support for apple M1

I could split this in multiple PRs instead.

File: configure [hunk 1]
```diff
@@ -1,15 +1,129 @@
...
+# Now run configure proper
+
+################################
+# Setup current directory where this script lives
+CFG_BIN="$( realpath "${BASH_SOURCE[0]}" )"
+CFG_ROOT_DIR="$( cd "$( dirname "${CFG_BIN}" )" && pwd )"
+
+CFG_BIN_DIR=$CFG_ROOT_DIR/$VIRTUALENV_DIR/bin
+
+# force relaunching under X86-64 archuitecture on macOS M1/ARM
```
Comment (JonoYang) [inline configure]: "archuitecture" should be "architecture"

File: src/packagedcode/pypi_setup_py.py [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+    'python_requires',
+    'requires',
+    'setup_requires',
+    'tests_require',
+    'url',
+    'version',
+}
+
+
+def is_setup_call(element):
```
Comment (JonoYang) [inline src/packagedcode/pypi_setup_py.py:50]: Just wondering if the `is_setup_call` function from `packagedcode/pypi.py` can be used here.

**Review by JonoYang — APPROVED** [review]
LGTM, just had a question about `is_setup_call`

File: src/packagedcode/pypi_setup_py.py (hunk 2 above)
Comment (pombredanne) [reply src/packagedcode/pypi_setup_py.py:50]: I actually created and copied is_setup_call() from packagedcode/pypi.py
They very closely related: I think the key difference is that this version works from a flat list of elements while the version in pypi.py works from a tree.
I added a note and a docstring. We could also remote the older code that I kept in pypi for now.

File: configure (hunk 1 above)
Comment (pombredanne) [reply configure]: Thanks! I pushed a commit to fix these.

**Comment by pombredanne** [comment]
@JonoYang Thanks!
See my comments from your review.
Please merge when this is passing if you thinks this is OK.
And if you do not mind to push afterwards a 31.0.0rc1 tag, this way we will get a release :)

**Comment by JonoYang** [comment]
@pombredanne Thanks, merging now!

**Pull request merged**